### PR TITLE
Updating `optimize` to add intervention builder that can handle optimizing the start time

### PR DESCRIPTION
**Pull request by anirban-chaudhuri**
- Added intervention builder templates to handle optimizing over intervention parameter value or intervention start time

### CODE REVIEW

File: tests/test_interfaces.py [hunk 1]
```diff
@@ -474,19 +474,15 @@ def test_optimize(model_fixture, start_time, end_time, num_samples):
...
-    intervened_params = optimize_kwargs["static_parameter_interventions"][
-        intervention_time[0]
-    ]
     result_opt = sample(
         model_url,
         end_time,
         logging_step_size,
         num_samples,
         start_time=start_time,
-        static_parameter_interventions={
```
Comment (SamWitty) [inline tests/test_interfaces.py:487]: Could you replace this `static_parameter_interventions=...` line with just `**optimize_kwargs`?

File: pyciemss/integration_utils/intervention_builder.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from typing import Dict, List
+
+import torch
+from chirho.interventional.ops import Intervention
+
+
+def param_value_objective(
```
Comment (SamWitty) [inline pyciemss/integration_utils/intervention_builder.py:7]: Can you add a return type here?

Comment (SamWitty) [reply pyciemss/integration_utils/intervention_builder.py:7]: Should be `Callable[[torch.Tensor], Dict[float, Dict[str, Intervention]]]`

File: pyciemss/integration_utils/intervention_builder.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                        param_name[count]: param_value[count](x[count].item())
+                    }
+                }
+            )
+        return static_parameter_interventions
+
+    return intervention_generator
+
+
+def start_time_objective(param_name: List[str], param_value: List[Intervention]):
```
Comment (SamWitty) [inline pyciemss/integration_utils/intervention_builder.py:32]: See above comment. Please add a return type.

File: tests/test_interfaces.py [hunk 4]
```diff
@@ -474,19 +474,15 @@ def test_optimize(model_fixture, start_time, end_time, num_samples):
         assert bounds_interventions[0][i] <= opt_policy[i]
         assert opt_policy[i] <= bounds_interventions[1][i]
 
-    intervention_time = list(optimize_kwargs["static_parameter_interventions"].keys())
-    intervened_params = optimize_kwargs["static_parameter_interventions"][
-        intervention_time[0]
-    ]
     result_opt = sample(
```
Comment (SamWitty) [inline tests/test_interfaces.py:477]: This `sample` doesn't appear to be using the results of the optimization.

**Review by SamWitty — CHANGES_REQUESTED** [review]
Looks good overall. Some minor requests for type changes. Please confirm that the optimization test is doing what we intend.

File: tests/test_interfaces.py (hunk 1 above)
Comment (anirban-chaudhuri) [reply tests/test_interfaces.py:487]: Fixed it to use the optimal policy

File: pyciemss/integration_utils/intervention_builder.py (hunk 2 above)
Comment (anirban-chaudhuri) [reply pyciemss/integration_utils/intervention_builder.py:7]: Done

File: tests/test_interfaces.py (hunk 4 above)
Comment (anirban-chaudhuri) [reply tests/test_interfaces.py:477]: Fixed it to use the optimal policy

File: pyciemss/integration_utils/intervention_builder.py (hunk 3 above)
Comment (anirban-chaudhuri) [reply pyciemss/integration_utils/intervention_builder.py:32]: Done.

File: tests/test_interfaces.py (hunk 4 above)
Comment (anirban-chaudhuri) [reply tests/test_interfaces.py:477]: Added test for optimizing start time

**Review by SamWitty — APPROVED** [review]

**Pull request merged**